ci/travis/setup.sh

```diff
@@ -7,7 +7,7 @@ set -e
 curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain=$TRAVIS_RUST_VERSION
 export PATH=$PATH:$HOME/.cargo/bin
 if [ "$TRAVIS_JOB_NAME" = "Minimum nightly" ]; then
-    rustup component add clippy
+    rustup component add clippy || cargo install --git https://github.com/rust-lang/rust-clippy/ --force clippy
```

Comment:
The code is fine, I just wanted to note that I only pinned nightlies in .travis.yml that have clippy.

Reply:
Correction: I actually think we should revert the change in .travis.yml because the master of clippy is not compatible with that rustc version.

Reply:
@konstin  I saw the issue and you're right ... we cannot build the latest clippy available with the min rust compiler supported by the project.
The major issue here is that instally clippy on `nightly-2019-06-22` does not work anymore even if it should be present (according to [static.rust-lang.org](https://static.rust-lang.org/dist/2019-06-22/)).
An example of build failure is available [here](https://travis-ci.com/macisamuele/pyo3/jobs/220053898)).

What would be your suggestion in this case?

My current idea would be to modify travis such that `Minimum nightly` task runs only tests and a new task (ie. `Lints`) will be added to uniquely run `cargo fmt` and `cargo clippy`.
Waiting for your feedback before continuing

Reply:
My suggestion is to revert the changes in `.travis.yml` and to `setup.sh`. That way ci should pass again. If that clippy version has bugs fixed in newer versions, we could bump the minimum nightly to e.g. `nightly-2019-08-01`, which also has clippy and rustfmt. (See https://rust-lang.github.io/rustup-components-history/)